src/Model/UnvalidatedOidcTokens.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Drenso\OidcBundle\Model;
+
+use DateTimeImmutable;
+use Drenso\OidcBundle\Enum\OidcTokenType;
+use Drenso\OidcBundle\Exception\OidcException;
+use stdClass;
+
+/**
+ * Contains the access and id tokens retrieved from OpenID authentication.
+ */
+class UnvalidatedOidcTokens
+{
+  protected readonly ?string $accessToken;
+  protected readonly ?string $idToken;
+  private ?DateTimeImmutable $expiry = null;
+  private ?string $refreshToken      = null;
+  /** @var string[]|null */
+  private ?array $scope = null;
+
+  /**
+   * @param stdClass|UnvalidatedOidcTokens $tokens
+   * @throws OidcException
+   */
+  public function __construct(object $tokens)
```

Comment:
```suggestion
  public function __construct(UnvalidatedOidcTokens|stdClass $tokens)
```

That way you only need to check the `instanceof UnvalidatedOidcTokens` and have an early return:

```
    if ($tokens instanceof UnvalidatedOidcTokens) {
      $this->accessToken = $tokens->getAccessToken();
      $this->idToken = $tokens->getAccessToken();
      $this->expiry = DateTimeImmutable::createFromInterface($tokens->getExpiry());
      $this->refreshToken = $tokens->getRefreshToken();
      $this->scope = $tokens->getScope();

      return;
    }
```

Reply:
`UnvalidatedOidcTokens is not an instance of `self` in the `OidcTokens` constructor, so our use case would not work:
```php
new OidcTokens(new UnvalidatedOidcTokens(...));
```
Instead, I used the combined type of `UnvalidatedOidcTokens|stdClass`, as you suggested earlier.

Reply:
As `self` is just an alias to `UnvalidatedOidcTokens`, so when using `public function __construct(self|stdClass $tokens)` in `UnvalidatedOidcTokens` you can still use (actually, you must) `public function __construct(UnvalidatedOidcTokens|stdClass $tokens)` in `OidcTokens` :) 

But no worries, I will adjust it.